RevenueCatUI/CustomerCenter/Data/SubscriptionInformation.swift

```diff
@@ -14,33 +14,178 @@
 //
 
 import Foundation
+import RevenueCat
 
+// swiftlint:disable nesting
 struct SubscriptionInformation {
 
-    let title: String
-    let durationTitle: String
-    let price: String
-    let expirationDateString: String?
+    let title: String?
+    let explanation: Explanation?
+    let durationTitle: String?
+    let price: PriceDetails
+    let expirationOrRenewal: ExpirationOrRenewal?
     let productIdentifier: String
 
     let willRenew: Bool
     let active: Bool
 
-    init(title: String,
-         durationTitle: String,
-         price: String,
-         expirationDateString: String?,
+    let store: Store?
+
+    init(title: String?,
+         explanation: Explanation?,
+         durationTitle: String?,
+         price: PriceDetails,
+         expirationOrRenewal: ExpirationOrRenewal?,
          willRenew: Bool,
          productIdentifier: String,
-         active: Bool
+         active: Bool,
+         store: Store?
     ) {
         self.title = title
+        self.explanation = explanation
         self.durationTitle = durationTitle
         self.price = price
-        self.expirationDateString = expirationDateString
+        self.expirationOrRenewal = expirationOrRenewal
         self.productIdentifier = productIdentifier
         self.willRenew = willRenew
         self.active = active
+        self.store = store
+    }
+
+    init?(entitlement: EntitlementInfo,
+          subscribedProduct: StoreProduct? = nil) {
+        // swiftlint:disable:next todo
+        // TODO: support non-consumables
+        let dateFormatter = DateFormatter()
```

Comment:
Maybe this can be an optional parameter (defaulted to `DateFormatter()`), or maybe the locale can be a parameter? I've noticed that some tests fail locally for me because my machine is not set to `en-US`. If we allow the test to specify a locale, the machine's locale has no influence on test results.

Reply:
right good point